thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomalydetection/performanceEvaluation/PrecisionByTimePerformanceEvaluation.java

```diff
@@ -52,13 +52,15 @@ public double evaluate() {
     }
 
     // take average of all the precisions
-    double avgPrecision = 0.0;
+    long totalDetectedAnomalyTimeLength = 0;
+    long totalDimensionOverlapTimeLength = 0;
     for(DimensionMap dimensions : dimensionToOverlapTimeLength.keySet()) {
-      if(dimensionToDetectedAnomalyTimeLength.get(dimensions) == 0l) {
-        continue;
-      }
-      avgPrecision += (double) dimensionToOverlapTimeLength.get(dimensions) / dimensionToDetectedAnomalyTimeLength.get(dimensions);
+      totalDetectedAnomalyTimeLength += dimensionToDetectedAnomalyTimeLength.get(dimensions);
+      totalDimensionOverlapTimeLength += dimensionToOverlapTimeLength.get(dimensions);
+    }
+    if (totalDetectedAnomalyTimeLength == 0) {
+      return Double.NaN;
     }
-    return avgPrecision / ((double) dimensionToDetectedAnomalyTimeLength.size());
+    return (double) totalDimensionOverlapTimeLength / totalDetectedAnomalyTimeLength;
```

Comment:
It is not required. The definition of F1 score is not changed. It is a different calculation strategy among different sub-metrics.